src/main/java/uk/gov/hmcts/reform/blobrouter/services/SasTokenGeneratorService.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.blobrouter.services;
+
+import com.azure.storage.blob.sas.BlobContainerSasPermission;
+import com.azure.storage.blob.sas.BlobServiceSasSignatureValues;
+import com.azure.storage.common.StorageSharedKeyCredential;
+import com.azure.storage.common.sas.SasProtocol;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration;
+import uk.gov.hmcts.reform.blobrouter.config.ServiceConfiguration.ServiceConfig;
+import uk.gov.hmcts.reform.blobrouter.exceptions.ServiceConfigNotFoundException;
+
+import java.time.OffsetDateTime;
+
+@EnableConfigurationProperties(ServiceConfiguration.class)
+@Service
+public class SasTokenGeneratorService {
+
+    private static final Logger log = LoggerFactory.getLogger(SasTokenGeneratorService.class);
+
+    private final StorageSharedKeyCredential storageSharedKeyCredential;
+    private final ServiceConfiguration serviceConfiguration;
+
+    public SasTokenGeneratorService(
+        StorageSharedKeyCredential storageSharedKeyCredential,
+        ServiceConfiguration serviceConfiguration
+    ) {
+        this.storageSharedKeyCredential = storageSharedKeyCredential;
+        this.serviceConfiguration = serviceConfiguration;
+    }
+
+    public String generateSasToken(String serviceName) {
+        log.info("SAS Token request received for service {} ", serviceName);
```

Comment:
Entry to service method theoretically can originate from anywhere. Perhaps we can just log the fact that we are `Generating SAS Token for {} service`?

Reply:
Updated the message